src/sagas/nanoContract.js

```diff
@@ -0,0 +1,115 @@
+import {
+  ncApi,
+  errors as hathorLibErrors
+} from '@hathor/wallet-lib';
+
+import { t } from 'ttag';
+
+import { getGlobalWallet } from "../modules/wallet";
+
+import {
+  addBlueprintInformation,
+  nanoContractRegisterError,
+  nanoContractRegisterSuccess,
+  types,
+} from '../actions';
+
+import nanoUtils from '../utils/nanoContracts';
+
+import { all, call, put, select, takeEvery } from 'redux-saga/effects';
+
+export const NANOCONTRACT_REGISTER_STATUS = {
+  LOADING: 'loading',
+  ERROR: 'error',
+  SUCCESS: 'success',
+};
+
+/**
+ * Process Nano Contract registration request.
+ * @param {{
+ *   payload: {
+ *     address: string,
+ *     ncId: string,
+ *   }
+ * }} action with request payload.
+ */
+export function* registerNanoContract({ payload }) {
+  const { address, ncId } = payload;
+
+  const blueprintsData = yield select((state) => state.blueprintsData);
+
+  const wallet = getGlobalWallet();
+  if (!wallet.isReady()) {
+    console.debug('Fail registering Nano Contract because wallet is not ready yet.');
+    yield put(nanoContractRegisterError(t`Wallet is not ready.`));
+    return;
+  }
+
+  const isRegistered = yield call([wallet.storage, wallet.storage.isNanoContractRegistered], ncId);
+  if (isRegistered) {
+    console.debug('Fail registering Nano Contract because it is already registered.');
+    yield put(nanoContractRegisterError(t`Nano contract is already registered.`));
+    return;
+  }
+
+  const isAddressMine = yield call([wallet, wallet.isAddressMine], address);
+  if (!isAddressMine) {
+    console.debug('Fail registering Nano Contract because address do not belongs to this wallet.');
+    yield put(nanoContractRegisterError(t`Address does not belong to the wallet.`));
+    return;
+  }
+
+  let tx = null;
+  try {
+    const response = yield call([wallet, wallet.getFullTxById], ncId);
+    tx = response.tx;
+  } catch (error) {
+    console.error('Error while registering Nano Contract.', error);
+    yield put(nanoContractRegisterError(t`Error while registering the nano contract.`));
+    return;
+  }
+
+  if (!nanoUtils.isNanoContractCreate(tx)) {
+    console.debug('Transaction is not a nano contract creation transaction.');
+    yield put(nanoContractRegisterError(t`Invalid nano contract creation transaction.`));
+    return;
+  }
+
+  let blueprintInformation = blueprintsData[tx.nc_blueprint_id];
+  if (!(blueprintInformation)) {
+    // We store the blueprint information in redux because it's used in some screens
+    // then we don't need to request it every time
```

Comment:
suggestion(non-blocking): this comment could also be copied on the `reducers` file, as an explanation on why to store that information there.

Reply:
Done